chainer/links/model/vision/googlenet.py

```diff
@@ -0,0 +1,401 @@
+                else:
+                    h = func(h)
+
+            if key in target_layers:
+                activations[key] = h
+                target_layers.remove(key)
+
+            if key == 'inception_4a':
+                inception_4a_cache = h
+            elif key == 'inception_4d':
+                inception_4d_cache = h
+
+        return activations
+
+    def extract(self, images, layers=['pool5'], size=(224, 224),
+                train=False, volatile=flag.OFF):
+        """Extracts all the feature maps of given images.
+
+        The difference of directly executing ``__call__`` is that
+        it directly accepts images as an input and automatically
+        transforms them to a proper variable. That is,
+        it is also interpreted as a shortcut method that implicitly calls
+        ``prepare`` and ``__call__`` functions.
+
+        Args:
+            images (iterable of PIL.Image or numpy.ndarray): Input images.
+            layers (list of str): The list of layer names you want to extract.
+            size (pair of ints): The resolution of resized images used as
+                an input of CNN. All the given images are not resized
+                if this argument is ``None``, but the resolutions of
+                all the images should be the same.
+            train (bool): If ``True``, Dropout runs in training mode.
+            volatile (~chainer.Flag): Volatility flag used for input variables.
+
+        Returns:
+            Dictionary of ~chainer.Variable: A directory in which
+            the key contains the layer name and the value contains
+            the corresponding feature map variable.
+
+        """
+
+        x = concat_examples([prepare(img, size=size) for img in images])
+        x = Variable(self.xp.asarray(x), volatile=volatile)
+        return self(x, layers=layers, train=train)
+
+    def predict(self, images, oversample=True):
+        """Computes all the probabilities of given images.
+
+        Args:
+            images (iterable of PIL.Image or numpy.ndarray): Input images.
+            oversample (bool): If ``True``, it averages results across
+                center, corners, and mirrors. Otherwise, it uses only the
+                center.
+
+        Returns:
+            ~chainer.Variable: Output that contains the class probabilities
+            of given images.
+
+        """
+
+        x = concat_examples([prepare(img, size=(256, 256)) for img in images])
+        if oversample:
+            x = imgproc.oversample(x, crop_dims=(224, 224))
+        else:
+            x = x[:, :, 16:240, 16:240]
+        # Set volatile option to ON to reduce memory consumption
+        x = Variable(self.xp.asarray(x), volatile=flag.ON)
+        y = self(x, layers=['prob'])['prob']
+        if oversample:
+            n = y.data.shape[0] // 10
+            y_shape = y.data.shape[1:]
+            y = reshape(y, (n, 10) + y_shape)
+            y = sum_function(y, axis=1) / 10
```

Comment:
I suppose that you decided to use `sum_function` to avoid overriding.
How about using `from chainer.functions.math import sum` and replacing it with `sum.sum(y, axis=1)`?